### DISCUSSION TITLE
2023 Roadmap

### DISCUSSION THREAD

**Discussion by akshayjshah** [connectrpc/.github#5] [Announcements]
# Roadmap for 2023

We've had a busy year! Since [announcing Connect][announcement] in the summer of 2022, we've shipped a stable implementation in [Go][connect-go], a release candidate in [TypeScript][connect-es], and alpha releases in [Swift][connect-swift] and [Kotlin][connect-kotlin]. We've also [extended the Connect protocol][get-protocol] to use `GET` requests for idempotent RPCs, integrated [Connect support into Envoy][envoy-example], and shipped a variety of interceptors and other add-ons. Our heartfelt thanks go out to everyone using Connect, submitting bug reports and pull requests, and chatting with us in [Slack](https://buf.build/links/slack) &mdash; your enthusiasm sustains this project!

Most of the work we're planning to do in the remainder of 2023 is continued refinement of existing projects. Those items are tracked in issues in each project. However, we are planning a few notable project-wide efforts:

* A stable release of the [TypeScript Connect implementation][connect-es]. ✅  Done!
* A more capable alternative to gRPC-Gateway. ✅  Done!
* A client-only C++ implementation.

[connect-go]: https://github.com/connectrpc/connect-go
[connect-es]: https://github.com/connectrpc/connect-es
[connect-swift]: https://github.com/connectrpc/connect-swift
[connect-kotlin]: https://github.com/connectrpc/connect-kotlin
[get-protocol]: https://buf.build/blog/introducing-connect-cacheable-rpcs
[envoy-example]: https://github.com/connectrpc/envoy-demo
[announcement]: https://buf.build/blog/connect-a-better-grpc

**Comment by akshayjshah** [comment]
As of today, [connect-es is stable](https://github.com/connectrpc/connect-es/releases/tag/v1.0.0)!

**Comment by nii236** [comment]
Hey @akshayjshah, super excited to see the updates on the C++ client implementation progress. Where can I find it?

**Reply by akshayjshah** [reply]
Hey @nii236! Sorry, completely missed this when you posted it. We're still working on this in the background, but community interest seems lower than Python, Rust, and Java. What sort of applications are you developing in C++?

**Reply by nii236** [reply]
Thanks @akshayjshah.

We're a game development company, using Unreal Engine which uses C++. 

Its way too late now unfortunately (the game studio has been wound up), but thanks for getting back to me regardless!

**Reply by akshayjshah** [reply]
Double apologies for the late reply, then! 😿

It's funny that you mention Unreal and game studios: that's exactly the environment we're thinking of targeting. When we're closer to a release, we'd love your opinion of connect-cc!

**Comment by rebello95** [comment]
I'd like to share that we recently launched [vanguard-go](https://github.com/connectrpc/vanguard-go) as a new repository in the Connect ecosystem! Vanguard is a powerful library for Go `net/http` servers that enables seamless transcoding between REST, gRPC, gRPC-Web, and Connect protocols without any code generation required. More details can be found in the repository's [README](https://github.com/connectrpc/vanguard-go/blob/main/README.md).

**Comment by akshayjshah** [comment]
New year, new roadmap! Closing this now that #16 is open.
